tools/autograd/derivatives.yaml

```diff
@@ -8,6 +8,9 @@
 #     Note that a single gradient entry can specify the gradient
 #     formula for multiple input names, by specifying a key
 #     "input1, input2" (see atan2 for an example).
+#   - An argument can be flagged as 'not_differentiable'. That means this
+#     argument cannot have a derivative defined and must not be used in any
+#     derivative calculations.
```

Comment:
This needs an explanation for how you handle "not_differentiable" vs not specifying the derivative.